atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,228 @@
+            .outerRule(GRADLE_PREPARE_TASK)
+            .around(MULTINODE_CASSANDRA_SETUP);
+
+    @Test
+    public void shouldRunTransactionsWithAllCassandraNodesRunningWithoutUnacceptableDelay()
+            throws InterruptedException, IOException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeWithAllNodesRunning = transactionEndTime - transactionStartTime;
+
+        assertThat("transactionTimeWithAllNodesRunning",
+                transactionTimeWithAllNodesRunning,
+                is(lessThan(MAX_CASSANDRA_NODES_RUNNING_MILLIS)));
+
+        String container = CASSANDRA_NODES.get(0);
+        checkNodetoolStatus(MULTINODE_CASSANDRA_SETUP.containers().container(container), "UN", 3);
+    }
+
+    @Test
+    public void shouldRunTransactionsAfterCassandraNodeIsShutDownWithoutUnacceptableDelay()
+            throws InterruptedException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+
+        String cassandraNodeToKill = getRandomCassandraNodeToShutdown();
+        killCassandraContainer(cassandraNodeToKill);
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeAfterNodeIsKilled = transactionEndTime - transactionStartTime;
+
+        startCassandraContainer(cassandraNodeToKill);
+
+        assertThat("transactionTimeAfterNodeIsKilled",
+                transactionTimeAfterNodeIsKilled,
+                is(lessThan(MAX_CASSANDRA_NODE_DOWN_MILLIS)));
+    }
+
+    private static DockerPort asPort(String node) {
+        return MULTINODE_CASSANDRA_SETUP.containers().container(node).port(TIMELOCK_SERVER_PORT);
+    }
+
+    private static <T> T createClientFor(Class<T> clazz, DockerPort port) {
+        String uri = port.inFormat("http://$HOST:$EXTERNAL_PORT");
+        return AtlasDbHttpClients.createProxy(Optional.absent(), uri, clazz);
+    }
+
+    private static HealthCheck<Container> toBeReady() {
+        return (container) -> {
+            TodoResource todos = createClientFor(TodoResource.class, container.port(TIMELOCK_SERVER_PORT));
+
+            return SuccessOrFailure.onResultOf(() -> {
+                todos.isHealthy();
+                return true;
+            });
+        };
+    }
+
+    private String getRandomCassandraNodeToShutdown() {
+        int index = ThreadLocalRandom.current().nextInt(CASSANDRA_NODES.size());
+        return CASSANDRA_NODES.get(index);
+    }
+
+    private static void killCassandraContainer(String containerName) {
```

Comment:
Is it worth moving all the cassandra node orchestration stuff below into a separate class, to separate the orchestration logic from our tests?